src/backend/distributed/transaction/transaction_management.c

```diff
@@ -335,7 +335,8 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 			 */
 			if (!IsMainDB && MainDBConnection != NULL)
 			{
-				RunCitusMainDBQuery(COMMIT_MANAGEMENT_COMMAND_2PC);
+				bool sendInXact = true;
```

Comment:
Should we still send COMMIT_MANAGEMENT_COMMAND_2PC / COMMIT even if didn't do RemoteTransactionBegin() in utility hook?